openlibrary/solr/data_provider.py

```diff
@@ -288,6 +289,31 @@ def get_editions_of_work(self, work):
         """
         raise NotImplementedError()
 
+    def get_solr_record(self, work_key: str) -> dict | None:
+        """
+        Fetches the record's information from Solr.
+        :param str work_key:type-prefixed key
+        :rtype: dict of str
+        """
+
+        response = requests.post(
+            get_solr_base_url() + '/query',
+            json={
+                "params": {
+                    "json.nl": "arrarr",
+                    "q": "key: %s" % work_key,
+                    "fl": [
+                        "trending_score_hourly_sum",
+                        'trending_score_hourly_*',
+                        "trending_score_daily_*",
+                        "trending_z_score",
+                    ],
+                    "sort": "trending_z_score desc",
+                }
+            },
+        )
+        return response.json()
```

Comment:
Instead of returning the full solr response, this would be a good place to define the interface of this method and return just the trending score data. This will also make it easier to override when we do the full-reindex flow.

Reply:
Excellent advice, and resolved accordingly. Thanks!